File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlDataTypeWithSize.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';
import { SqlDataSize } from './SqlDataSize';

export class SqlDataTypeWithSize implements ISqlExpression {
    SqlType: SqlType = SqlType.DataTypeWithSize;
    Span: TextSpan = new TextSpan();
    DataTypeName: string = '';
    Size: SqlDataSize = new SqlDataSize();

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_DataTypeWithSize(this);
    }

    ToSql(): string {
        return this.DataTypeName + this.Size.ToSql();
    }
} 

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlDataTypeWithSize.cs
using T1.Standard.IO;

namespace T1.SqlSharp.Expressions;

public class SqlDataTypeWithSize : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.DataTypeWithSize;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_DataTypeWithSize(this);
    }

    public string DataTypeName { get; set; } = string.Empty;
    public SqlDataSize Size { get; set; } = new();
    public string ToSql()
    {
        var sql = new IndentStringBuilder();
        sql.Write(DataTypeName);
        sql.Write(Size.ToSql());
        return sql.ToString();
    }
}